### PR TITLE
added length checking for EmbedBuilder

### PR DESCRIPTION
**Pull request by MaxGrosshandler**

### CODE REVIEW

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 @SuppressWarnings({"WeakerAccess", "unused"})
 public class EmbedBuilder {
+
```
Comment (queer) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: why do you do this to me

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 2]
```diff
@@ -139,12 +140,22 @@ public EmbedBuilder field(@Nonnull final Field field) {
     }
     
     public Embed build() {
+        int len = 0;
```
Comment (queer) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: maKE THINGS FINAL AAAAAAAA

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 3]
```diff
@@ -162,16 +177,31 @@ public Embed build() {
...
+            for (Field item : fields) {
+                if (item.name().length() > 256){
+                throw new IllegalStateException("Field name exceeds 256 characters!");
+            }
+            if (item.value().length() > 1024){
+                throw new IllegalStateException("Field value exceeds 1024 characters!");
+            }
+                len += item.name().length();
+                len += item.value().length();
+}
```
Comment (queer) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: wtf is this formatting?

Comment (natanbc) [reply src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: smells like tabs

**Comment by MaxGrosshandler** [comment]
fixed changes as requesting

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 4]
```diff
@@ -162,16 +176,33 @@ public Embed build() {
...
         if(author != null) {
+            if(author.name().length() > 256){
+                throw new IllegalStateException("Author's name exceeds 256 characters!");
+            }
+            len += author.name().length();
             builder.author(author);
         }
         if(fields.isEmpty()) {
-            builder.fields(Collections.emptyList());
+        	builder.fields(Collections.emptyList());
```
Comment (natanbc) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: should be 4 spaces

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 5]
```diff
@@ -139,12 +139,22 @@ public EmbedBuilder field(@Nonnull final Field field) {
     }
     
     public Embed build() {
+    	int len = 0;
```
Comment (natanbc) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: wrong indentation

File: src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java [hunk 6]
```diff
@@ -162,16 +176,33 @@ public Embed build() {
...
+                    throw new IllegalStateException("Field value exceeds 1024 characters!");
+                }
+                len += field.name().length();
+                len += field.value().length();
+            }
             builder.fields(fields);
         }
+         if(len > 6000){
+             throw new IllegalStateException("Total embed length exceeds 6000 characters!");
+            }
```
Comment (natanbc) [inline src/main/java/com/mewna/catnip/entity/builder/EmbedBuilder.java]: wrong indentation

**Comment by MaxGrosshandler** [comment]
God willing, final changes

**Pull request merged**